### PR TITLE
fix: make authorization module dynamic in order to pass permission provider

### PR DESCRIPTION
**Pull request by edgarmueller**
Allow `AuthorizationModule` to be initialized once via `register` and a `permissionProvider` options which expects a type. Use registered module via `use` to get reference to the initalized module.

### CODE REVIEW

File: packages/nestjs-authorization/src/authorization.module.ts [hunk 1]
```diff
@@ -1,10 +1,33 @@
...
-  imports: [],
-  providers: [ConditionService, PermissionService],
-  exports: [ConditionService, PermissionService],
-})
-export class AuthorizationModule {}
+@Module({})
+export class AuthorizationModule {
+  private static module: DynamicModule;
+
+  static register({
```
Comment (ottovw) [inline packages/nestjs-authorization/src/authorization.module.ts:11]: what about registerAsync?

File: example-apps/my-nest-project/src/app.module.ts [hunk 2]
```diff
@@ -10,9 +10,10 @@ import { AppService } from './app.service';
 import { IsAuthor } from './is-author.condition';
 import { AuthnGuard } from './authn.guard';
 import { InMemoryPermissionProvider } from './in-memory-permission.provider';
+import { UsersModule } from './users/users.module';
 
 @Module({
-  imports: [AuthorizationModule],
+  imports: [AuthorizationModule.use(), UsersModule],
```
Comment (ottovw) [inline example-apps/my-nest-project/src/app.module.ts:16]: why is this necessary? Don't know of any other Module which does this. Is this really necessary?

File: packages/nestjs-authorization/.babelrc [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{
```
Comment (ottovw) [inline packages/nestjs-authorization/.babelrc:1]: please rebase before merge

**Review by ottovw — CHANGES_REQUESTED** [review]

File: packages/nestjs-authorization/src/authorization.module.ts (hunk 1 above)
Comment (edgarmueller) [reply packages/nestjs-authorization/src/authorization.module.ts:11]: Why would we need that?

File: packages/nestjs-authorization/.babelrc (hunk 3 above)
Comment (edgarmueller) [reply packages/nestjs-authorization/.babelrc:1]: What are you referring to?

File: example-apps/my-nest-project/src/app.module.ts (hunk 2 above)
Comment (edgarmueller) [reply example-apps/my-nest-project/src/app.module.ts:16]: I'll check again

Comment (edgarmueller) [reply example-apps/my-nest-project/src/app.module.ts]: Removed `use` method, instead we could rely on re-exporting in userland instead. Do you prefer that?

File: packages/nestjs-authorization/.babelrc (hunk 3 above)
Comment (ottovw) [reply packages/nestjs-authorization/.babelrc:1]: these are changes of other PR / branch

File: packages/nestjs-authorization/src/authorization.module.ts (hunk 1 above)
Comment (ottovw) [reply packages/nestjs-authorization/src/authorization.module.ts:9]: E.g. if your permission provider has dependencies. e.g. ConfigService.

Other modules offer something like this:

```typescript
TypeOrmModule.forRootAsync({
      imports: [ConfigModule],
      useFactory: (config: ConfigService) => config.get('database'),
      inject: [ConfigService],
    }),
```

**Review by ottovw — CHANGES_REQUESTED** [review]

File: example-apps/my-nest-project/src/app.module.ts (hunk 2 above)
Comment (ottovw) [reply example-apps/my-nest-project/src/app.module.ts]: I don't understand consequences, let's discuss in person

**Review by ottovw — APPROVED** [review]
👍
